src/Nethermind/Nethermind.AccountAbstraction/Source/UserOperationPool.cs

```diff
@@ -65,16 +73,33 @@ public class UserOperationPool : IUserOperationPool
             _userOperationSortedPool = userOperationSortedPool;
             _userOperationSimulator = userOperationSimulator;
             _simulatedUserOperations = simulatedUserOperations;
+            _peerManager = peerManager;
+            _p2pProtocolHandler = protocolHandler;
 
             blockTree.NewHeadBlock += OnNewBlock;
             _userOperationSortedPool.Inserted += UserOperationInserted;
             _userOperationSortedPool.Removed += UserOperationRemoved;
+
         }
         
         private void UserOperationInserted(object? sender, SortedPool<UserOperation, UserOperation, Address>.SortedPoolEventArgs e)
         {
             UserOperation userOperation = e.Key;
             SimulateAndAddToPool(userOperation, _blockTree.Head!.Header);
+            BroadcastToCompatiblePeers(userOperation, _peerManager.ConnectedPeers, _p2pProtocolHandler);
+        }
+
+        private void BroadcastToCompatiblePeers(UserOperation userOperation, IReadOnlyCollection<Peer> peers,
+            IP2PProtocolHandler protocolHandler)
+        {
+            Capability? aaCapabiltiy = new Capability(Protocol.AA, 0);
+            foreach (var peer in peers)
+            {
+                if (protocolHandler.HasAgreedCapability(aaCapabiltiy))
```

Comment:
surely we want to check outside the function if our protocol can handle the capability, and only call its logic if yes